docs/source/nodes_and_pipelines/pipeline_registry.md

```diff
@@ -0,0 +1,73 @@
+# The pipeline registry
+
+Projects generated using Kedro 0.17.2 or later define their pipelines in `src/<package_name>/pipeline_registry.py`. The `pipeline_registry` module must contain a top-level `register_pipelines()` function that returns a mapping from pipeline names to [`Pipeline`](/kedro.pipeline.Pipeline) objects. For example, the pipeline registry in the [Kedro starter for the completed spaceflights tutorial](https://github.com/kedro-org/kedro-starters/tree/main/spaceflights) could define the following `register_pipelines()` function that exposes the data processing pipeline, the data science pipeline, and a third, default pipeline that combines both of the aforementioned pipelines:
+
+```python
+def register_pipelines() -> Dict[str, Pipeline]:
+    """Register the project's pipelines.
+
+    Returns:
+        A mapping from pipeline names to ``Pipeline`` objects.
+    """
+    data_processing_pipeline = dp.create_pipeline()
+    data_science_pipeline = ds.create_pipeline()
+
+    return {
+        "__default__": data_processing_pipeline + data_science_pipeline,
+        "data_processing": data_processing_pipeline,
+        "data_science": data_science_pipeline,
+    }
+```
+
+```{note}
+The order in which you add the pipelines together is not significant (`data_science_pipeline + data_processing_pipeline` would produce the same result), since Kedro automatically detects the data-centric execution order for all the nodes in the resulting pipeline.
+```
+
+## Pipeline autodiscovery
+
+In the above example, you need to update the `register_pipelines()` function whenever you create a pipeline that should be returned as part of the project's pipelines. Since Kedro 0.18.3, you can achieve the same result with less code using `find_pipelines()`[/kedro.framework.project.find_pipelines]:
```

Comment:
Following the above comment, this would then link to the file in the `main` branch.

Reply:
@AntonyMilneQB Sorry, I don't think I follow what needs to change here.

Reply:
So my reading of the documentation as you have it is as follows.

> For example, the pipeline registry in the [Kedro starter for the completed spaceflights tutorial](https://github.com/kedro-org/kedro-starters/tree/main/spaceflights) could define the following `register_pipelines()` function that exposes the data processing pipeline, the data science pipeline, and a third, default pipeline that combines both of the aforementioned pipelines... [example code as it was in 0.18.2]

and then you say

> Since Kedro 0.18.3, you can achieve the same result with less code using `find_pipelines()`[/kedro.framework.project.find_pipelines]: [example code as it is in 0.18.3]

So my thinking was the first of these examples should link to the pipeline_registry.py file as it is in 0.18.2. Otherwise when I look at kedro's docs for 0.18.3, the link you follow to the `register_pipelines` function (that currently goes to `main`) will point to code using `find_pipelines` that doesn't look like the example code directly below the text. The example code you have there is copied and pasted from the spaceflights 0.18.2 starter. Then I was also thinking that the "example code as it is in 0.18.3" should point to `main` so it's always linked to the latest version of the `register_pipelines` file.

Does that make sense? Ultimately I don't think the kedro versions really matter here; what matters is the idea that doing the manual pipeline definitions of the second example is equivalent to the automatic `find_pipelines` in the second example (which is clear I think). I just thought that you were trying to link to the files that had exactly those code snippets, and the one that points to `main` was actually meant to point to 0.18.2.